test/external_array_data_creation.cpp

```diff
@@ -259,5 +259,46 @@ namespace sparrow::test
         arr.release = &release_arrow_array;
     }
 
+    void fill_schema_and_array_for_run_end_encoded(
```

Comment:
Can you put this in arrow_array_schema_factory.hpp and arrow_array_schema_proxy_factory.hpp

Reply:
for non of the other arrays we did that `external_array_data_creation` has similar code for a bunch of other layouts...

Reply:
I know but at a moment we will have to use such function to create data ex nihilo

Reply:
I dont know what ex nihilo means =)

Reply:
I mean at some point we will have to create C structures when the user will want to create an array from some buffers. We want nice functions to create c structures yes.